data-structures/trie/trie.py

```diff
@@ -0,0 +1,40 @@
+class trienode:
+	def __init__(self,character):
+		self.character = character
+		self.nextnodes = {}
+		self.isEnd = False
+		self.dataNode = None
+	
+class trie: # use case: username/password
+	def __init__(self):
+		self.start = trienode('')
+	def insert(self,id,data): # returns true on successful insertion (data == password)
+		temp = self.start
+		for x in id:
+			temp.nextnodes[x] = temp.nextnodes.get(x,trienode(x))
+			temp = temp.nextnodes[x]
+		if temp.isEnd == True:
+			print("ID already Exists!!")
+			return False
+		temp.isEnd =True
+		temp.dataNode = data
+		return True
+	def findNode(self,id): # returns false if node doesn't exist and true, node if it does
+		temp = self.start
+		for x in id:
```

Comment:
Please use appropriate name for variable `x`

Reply:
Done